Frameworks/CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -707,11 +707,14 @@ CF_INLINE uintptr_t __CFISAForTypeID(CFTypeID typeID) {
 CF_INLINE bool __CF_IsBridgedObject(CFTypeRef obj) {
     CFRuntimeBase* object = (CFRuntimeBase*)obj;
     if (!object || 
-        (object->_cfisa == 0) || 
-        (object->_cfisa == (uintptr_t)(&_OBJC_CLASS__NSCFString)) || 
+        (object->_cfisa == 0)) {
+        return false;
+    }
+
+    if ((object->_cfisa == (uintptr_t)(&_OBJC_CLASS__NSCFString)) || 

```

Comment:
This is not actually a performance optimization (although it may be useful in that sense too). This is because these three classes in particular are used at static init time and these checks may happen *before* they are registered as bridge classes due to the non determinism in static init.

---
In reply to: [69767697](https://github.com/Microsoft/WinObjC/pull/609#discussion_r69767697) [](ancestors = 69767697)